SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -48,6 +48,23 @@ ASpatialFunctionalTest::ASpatialFunctionalTest()
 	PreparationTimeLimit = 30.0f;
 	bReadyToSpawnServerControllers = false;
 	CachedTestResult = EFunctionalTestResult::Default;
+
+	bIsStandaloneTest = false;
+	GeneratedTestMap = nullptr;
+	bIsGeneratingMap = false;
+}
+
+ASpatialFunctionalTest::ASpatialFunctionalTest(const EMapCategory MapCiCategory, const int32 NumberOfClients /*=1*/,
+											   const FVector& InTestPositionInWorld /*=FVector::ZeroVector*/)
+	: ASpatialFunctionalTest()
+{
+	bIsStandaloneTest = true;
+	TestPositionInWorld = InTestPositionInWorld;
+
+	GeneratedTestMap = NewObject<UGeneratedTestMap>();
+	GeneratedTestMap->Init(MapCiCategory, this->GetClass()->GetName());
```

Comment:
Why not use the overridden constructor?

Reply:
I'm a ue noob - I couldn't find a way to pass parameters to NewObject. Or are you suggesting forgoing UObject-ness and working around what we're using that for?

Reply:
`NewObject<UGeneratedTestMap>(/* how about putting parameters in here ; )*/);`

Reply:
That doesn't compile! Definitions of NewObject: 
	T* NewObject(UObject* Outer, const UClass* Class, FName Name = NAME_None, EObjectFlags Flags = RF_NoFlags, UObject* Template = nullptr, bool bCopyTransientsFromClassDefaults = false, FObjectInstancingGraph* InInstanceGraph = nullptr, UPackage* ExternalPackage = nullptr)

	T* NewObject(UObject* Outer = (UObject*)GetTransientPackage())

	T* NewObject(UObject* Outer, FName Name, EObjectFlags Flags = RF_NoFlags, UObject* Template = nullptr, bool bCopyTransientsFromClassDefaults = false, FObjectInstancingGraph* InInstanceGraph = nullptr)

Reply:
Gah, I swore NewObject had smart pointer like semantics were you can forward constructor arguments, but apparently not : /.

Reply:
Added a static method that calls both newobject and init (and init is private, yay!)